src/libcephsqlite.cc

```diff
@@ -1,473 +1,576 @@
+static int Unlock(sqlite3_file *pFile, int eLock)
 {
-  CephFile *p = (CephFile*)pFile;
-  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << "write(" << p->name << ", " << std::dec << iAmt << ", " << std::dec << iOfst << ")" << std::endl;
-  libradosstriper::RadosStriper *rs = get_radosstriper(get_db_name(p->name).c_str());
+  auto f = (cephsqlite_file*)pFile;
+  auto start = ceph::coarse_mono_clock::now();
+  df(5) << std::hex << eLock << dendl;
 
-  /* ceph::bufferlist buffer pointers are all char* */
-  char *b = reinterpret_cast<char*>(const_cast<void*>(zBuf));
-  ceph::bufferlist bl = ceph::bufferlist::static_from_mem(b, iAmt);
-
-  if (rs->write(p->name, bl, iAmt, iOfst) < 0)
-    return SQLITE_IOERR;
+  auto& lock = f->lock;
+  ceph_assert(lock == SQLITE_LOCK_NONE || (lock > SQLITE_LOCK_NONE && f->io.rs->is_locked()));
+  ceph_assert(lock >= eLock);
+  if (eLock <= SQLITE_LOCK_NONE && SQLITE_LOCK_NONE < lock) {
+    if (int rc = f->io.rs->unlock(); rc < 0) {
+      df(5) << "failed: " << rc << dendl;
+      return SQLITE_IOERR;
+    }
+  }
 
+  lock = eLock;
+  auto end = ceph::coarse_mono_clock::now();
+  getdata(f->vfs).logger->tinc(P_OPF_UNLOCK, end-start);
   return SQLITE_OK;
 }
 
-/*
-** Truncate a file. This is a no-op for this VFS (see header comments at
-** the top of the file).
-*/
-static
-int _cephsqlite3_Truncate(sqlite3_file *pFile, sqlite_int64 size)
+static int CheckReservedLock(sqlite3_file *pFile, int *pResOut)
 {
-  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << std::endl;
-  CephFile *p = (CephFile*)pFile;
-  libradosstriper::RadosStriper *rs = get_radosstriper(get_db_name(p->name).c_str());
+  auto f = (cephsqlite_file*)pFile;
+  auto start = ceph::coarse_mono_clock::now();
+  df(5) << dendl;
 
-  if (rs->trunc(p->name, size) != 0)
-    return SQLITE_IOERR;
+  auto& lock = f->lock;
+  if (lock > SQLITE_LOCK_SHARED) {
+    *pResOut = 1;
+  }
 
+  df(10);
+  f->io.rs->printlockers(*_dout);
+  *_dout << dendl;
+
+  *pResOut = 0;
+  auto end = ceph::coarse_mono_clock::now();
+  getdata(f->vfs).logger->tinc(P_OPF_CHECKRESERVEDLOCK, end-start);
   return SQLITE_OK;
 }
 
-/*
-** Sync the contents of the file to the persistent media.
-*/
-static
-int _cephsqlite3_Sync(sqlite3_file *pFile, int flags)
+static int Close(sqlite3_file *pFile)
 {
-  // std::cerr << std::hex << pthread_self() << ":" << __FUNCTION__ << "::" << std::endl;
+  auto f = (cephsqlite_file*)pFile;
+  auto start = ceph::coarse_mono_clock::now();
+  df(5) << dendl;
+  f->~cephsqlite_file();
+  auto end = ceph::coarse_mono_clock::now();
+  getdata(f->vfs).logger->tinc(P_OPF_CLOSE, end-start);
   return SQLITE_OK;
 }
 
-
-/*
-** Write the size of the file in bytes to *pSize.
-*/
-static
-int _cephsqlite3_FileSize(sqlite3_file *pFile, sqlite_int64 *pSize)
+static int Read(sqlite3_file *pFile, void *zBuf, int iAmt, sqlite_int64 iOfst)
```

Comment:
we don't use camelCase naming convention, could use names like `amount`, `offset`? probably we don't need to use the identical names used by the function signatures defined in `sqlite3_io_methods` ?